2.evaluate_models/classify_genotypes/nbconverted/plate_4_sirna_genotype_probabilities.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Evaluate model performance siRNA-treated single cells
+# Genotype probabilies of siRNA-treated single cell data are computed to evaluate model perormance.
+# This is performed on plate 4.
+
+# In[1]:
+
+
+import pathlib
+
+import numpy as np
+import pandas as pd
+from joblib import load
+
+# ## Find the root of the git repo on the host system
+
+# In[2]:
+
+
+# Get the current working directory
+cwd = pathlib.Path.cwd()
+
+if (cwd / ".git").is_dir():
+    root_dir = cwd
+
+else:
+    root_dir = None
+    for parent in cwd.parents:
+        if (parent / ".git").is_dir():
+            root_dir = parent
+            break
+
+# Check if a Git root directory was found
+if root_dir is None:
+    raise FileNotFoundError("No Git root directory found.")
+
+
+# ## Define paths
+
+# ### Input
+
+# In[3]:
+
+
+plate4df_path = pathlib.Path(root_dir / "nf1_painting_repo/3.processing_features/data/single_cell_profiles/Plate_4_sc_feature_selected.parquet").resolve(strict=True)
+plate4df = pd.read_parquet(plate4df_path)
+
+data_path = pathlib.Path(root_dir / "1.train_models/classify_genotypes/data").resolve(strict=True)
+le = load(f"{data_path}/trained_nf1_model_label_encoder.joblib")
+model = load(f"{data_path}/trained_nf1_model.joblib")
+
+# Set the seed
```

Comment:
Just curious: do the predicted probability values change every time you apply the model (kinda like UMAP embeddings), which is why you need to set a seed?

Reply:
Oh no, the logistic regression is already parameterized. That was unused code, it's removed now!